src/main/java/redis/clients/jedis/MultiNodePipelineBase.java

```diff
@@ -38,53 +38,38 @@
 import redis.clients.jedis.search.schemafields.SchemaField;
 import redis.clients.jedis.timeseries.*;
 import redis.clients.jedis.util.IOUtils;
+import redis.clients.jedis.util.JedisThreadPool;
 import redis.clients.jedis.util.KeyValue;
 
 public abstract class MultiNodePipelineBase implements PipelineCommands, PipelineBinaryCommands,
     RedisModulePipelineCommands, Closeable {
 
   private static final Logger log = LoggerFactory.getLogger(MultiNodePipelineBase.class);
 
+  public static volatile int MULTI_NODE_PIPELINE_SYNC_WORKERS = 3;
+
   private final Map<HostAndPort, Queue<Response<?>>> pipelinedResponses;
   private final Map<HostAndPort, Connection> connections;
   private volatile boolean syncing = false;
 
   private final CommandObjects commandObjects;
   private GraphCommandObjects graphCommandObjects;
 
-  /**
-   * The following are the default parameters for the multi node pipeline executor
-   * Since Redis query is usually a slower IO operation (requires more threads),
-   * so we set DEFAULT_CORE_POOL_SIZE to be the same as the core
-   */
-  private static final long DEFAULT_KEEPALIVE_TIME_MS = 60000L;
-  private static final int DEFAULT_BLOCKING_QUEUE_SIZE = Protocol.CLUSTER_HASHSLOTS;
-  private static final int DEFAULT_CORE_POOL_SIZE = Runtime.getRuntime().availableProcessors();
-  private static final int DEFAULT_MAXIMUM_POOL_SIZE = Runtime.getRuntime().availableProcessors() * 2;
-  private static ExecutorService executorService = JedisThreadPoolBuilder.pool()
-      .setCoreSize(DEFAULT_CORE_POOL_SIZE)
-      .setMaxSize(DEFAULT_MAXIMUM_POOL_SIZE)
-      .setKeepAliveMillSecs(DEFAULT_KEEPALIVE_TIME_MS)
-      .setThreadNamePrefix("jedis-multi-node-pipeline")
-      .setWorkQueue(new ArrayBlockingQueue<>(DEFAULT_BLOCKING_QUEUE_SIZE)).build();
+  private final ExecutorService executorService = getThreadPool();
```

Comment:
We need to define `executorService ` as a `static` variable to avoid multiple creation and destruction.